pkg/kv/mem/store.go

```diff
@@ -66,14 +61,14 @@ func (s *Store) Get(_ context.Context, partitionKey, key []byte) (*kv.ValueWithP
 	s.mu.RLock()
 	defer s.mu.RUnlock()
 
-	combinedKey := combinedKey(partitionKey, key)
-	value, ok := s.m[combinedKey]
+	sKey := encodeKey(key)
+	value, ok := s.m[string(partitionKey)][sKey]
```

Comment:
if `s.m` is missing the `partitionKey`, you'll get the default value of `PartitionMap` which is `nil` and your code will panic. I wonder how did no test find this. I think we should add a test for non existing partition.

Reply:
It doesn't panic, this specific scenario is tested, and this is similar to https://github.com/treeverse/lakeFS/pull/3632#discussion_r917385528 comment